pkg/dataobj/dataobj.go

```diff
@@ -181,6 +183,45 @@ func NewBuilder(cfg BuilderConfig, bucket objstore.Bucket, tenantID string) (*Bu
 	}, nil
 }
 
+// FromExisting updates this builder with content from an existing data object, replicating all the state like stream IDs and logs.
+func (b *Builder) FromExisting(f io.ReadSeeker) error {
```

Comment:
I'm not a massive fan of this method, and I think it could be made more efficient. I want to test it out before committing to any improvements here though.